server/src/main/java/org/uiautomation/ios/instruments/commandExecutor/CURLIAutomationCommandExecutor.java

```diff
@@ -112,7 +116,24 @@ private void getResponse(HttpServletRequest request, HttpServletResponse respons
       if (request.getInputStream() != null) {
         StringWriter writer = new StringWriter();
         IOUtils.copy(request.getInputStream(), writer, "UTF-8");
-        String json = writer.toString();
+        String json = null;
+        String sessionUUID = null;
+        String jsonConcatenated = "";
+        if (writer.toString().endsWith(PARTIAL_IDENTIFIER)){
+            sessionUUID = writer.toString().substring(0, UUID_LENGTH);
+            if (jsonMap.get(sessionUUID) != null){
+                jsonConcatenated = (String)jsonMap.get(sessionUUID);
+            }
+            jsonConcatenated =  jsonConcatenated.concat(writer.toString().substring(UUID_LENGTH, writer.toString().length() - PARTIAL_IDENTIFIER.length()));
```

Comment:
This suggested change will break the fix as it is chopping off an extra 36 chars.